components/feature/downloads/src/test/java/mozilla/components/feature/downloads/AbstractFetchDownloadServiceTest.kt

```diff
@@ -38,9 +39,11 @@ class AbstractFetchDownloadServiceTest {
     @Before
     fun setup() {
         initMocks(this)
-        service = spy(object : AbstractFetchDownloadService(broadcastManager) {
+        service = spy(object : AbstractFetchDownloadService() {
             override val httpClient = client
         })
+        doReturn(broadcastManager).`when`(service).broadcastManager
+        doReturn(testContext).`when`(service).context
     }
 
     @Test
```

Comment:
I wonder if we could add some more tests to this eventually. With a mocked `Client` you can test all those cases where exceptions are thrown or if the download fails. How about we file a follow-up to test more cases here?

Reply:
Filed #3917